### ISSUE TITLE
How can i use with EfCore Shadow Property

### ISSUE THREAD

**Issue by NetChenQian** [linq2db/linq2db#3169]
### Description

When i use with EfCore Shadow Property, The following error occurs：
**LinqToDB.Linq.LinqException:“'Property(p, "IsDeleted")' cannot be converted to SQL.”**

![image](https://user-images.githubusercontent.com/29855154/128753579-aa8bca54-d6bb-4b6d-887d-809e8490eec5.png)

I saw that this problem was fixed in version 1.5.2 [LinqToDB dosen't work well with EF Core soft delete.](https://github.com/linq2db/linq2db/issues/1626#issuecomment-469678181), but it didn't seem to work in version 5.4  [linq2db.EntityFrameworkCore 5.4.0](https://www.nuget.org/packages/linq2db.EntityFrameworkCore/5.4.0) after testing 

### Stack Trace

```
LinqToDB.Linq.LinqException
  HResult=0x80131500
  Message='Property(p, "IsDeleted")' cannot be converted to SQL.
  Source=linq2db
  StackTrace:
   at LinqToDB.Linq.Builder.ExpressionBuilder.ConvertToSql(IBuildContext context, Expression expression, Boolean unwrap, ColumnDescriptor columnDescriptor, Boolean isPureExpression)
   at LinqToDB.Linq.Builder.ExpressionBuilder.ConvertPredicate(IBuildContext context, Expression expression)
   at LinqToDB.Linq.Builder.ExpressionBuilder.ConvertToSql(IBuildContext context, Expression expression, Boolean unwrap, ColumnDescriptor columnDescriptor, Boolean isPureExpression)
   at LinqToDB.Linq.Builder.ExpressionBuilder.ConvertPredicate(IBuildContext context, Expression expression)
   at LinqToDB.Linq.Builder.ExpressionBuilder.BuildSearchCondition(IBuildContext context, Expression expression, List`1 conditions)
   at LinqToDB.Linq.Builder.ExpressionBuilder.BuildSearchCondition(IBuildContext context, Expression expression, List`1 conditions)
   at LinqToDB.Linq.Builder.ExpressionBuilder.BuildWhere(IBuildContext parent, IBuildContext sequence, LambdaExpression condition, Boolean checkForSubQuery, Boolean enforceHaving)
   at LinqToDB.Linq.Builder.WhereBuilder.BuildMethodCall(ExpressionBuilder builder, MethodCallExpression methodCall, BuildInfo buildInfo)
   at LinqToDB.Linq.Builder.MethodCallBuilder.BuildSequence(ExpressionBuilder builder, BuildInfo buildInfo)
   at LinqToDB.Linq.Builder.ExpressionBuilder.BuildSequence(BuildInfo buildInfo)
   at LinqToDB.Linq.Builder.TableBuilder.ApplyQueryFilters(ExpressionBuilder builder, BuildInfo buildInfo, MemberInfo memberInfo, TableContext tableContext)
   at LinqToDB.Linq.Builder.TableBuilder.BuildSequence(ExpressionBuilder builder, BuildInfo buildInfo)
   at LinqToDB.Linq.Builder.ExpressionBuilder.BuildSequence(BuildInfo buildInfo)
   at LinqToDB.Linq.Builder.OrderByBuilder.BuildMethodCall(ExpressionBuilder builder, MethodCallExpression methodCall, BuildInfo buildInfo)
   at LinqToDB.Linq.Builder.MethodCallBuilder.BuildSequence(ExpressionBuilder builder, BuildInfo buildInfo)
   at LinqToDB.Linq.Builder.ExpressionBuilder.BuildSequence(BuildInfo buildInfo)
   at LinqToDB.Linq.Builder.TakeSkipBuilder.BuildMethodCall(ExpressionBuilder builder, MethodCallExpression methodCall, BuildInfo buildInfo)
   at LinqToDB.Linq.Builder.MethodCallBuilder.BuildSequence(ExpressionBuilder builder, BuildInfo buildInfo)
   at LinqToDB.Linq.Builder.ExpressionBuilder.BuildSequence(BuildInfo buildInfo)
   at LinqToDB.Linq.Builder.TakeSkipBuilder.BuildMethodCall(ExpressionBuilder builder, MethodCallExpression methodCall, BuildInfo buildInfo)
   at LinqToDB.Linq.Builder.MethodCallBuilder.BuildSequence(ExpressionBuilder builder, BuildInfo buildInfo)
   at LinqToDB.Linq.Builder.ExpressionBuilder.BuildSequence(BuildInfo buildInfo)
   at LinqToDB.Linq.Builder.ExpressionBuilder.Build[T]()
   at LinqToDB.Linq.Query`1.CreateQuery(IDataContext dataContext, Expression expr)
   at LinqToDB.Linq.Query`1.GetQuery(IDataContext dataContext, Expression& expr)
   at LinqToDB.Linq.ExpressionQuery`1.GetQuery(Expression& expression, Boolean cache)
   at LinqToDB.Linq.ExpressionQuery`1.<GetForEachAsync>d__19.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable.ConfiguredTaskAwaiter.GetResult()
   at LinqToDB.AsyncExtensions.<ToListAsync>d__8`1.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at ConsoleApp3.Program.<Main>d__0.MoveNext() in H:\Code\linq2db.EntityFrameworkCore\ConsoleApp3\Program.cs:line 30
```

### Test Code

```c#
using System;
using System.Linq;
using System.Threading.Tasks;
using LinqToDB.EntityFrameworkCore.BaseTests;
using Microsoft.EntityFrameworkCore;
using Microsoft.Extensions.Configuration;
using LinqToDB;
using LinqToDB.EntityFrameworkCore;

namespace ConsoleApp3
{
	static class Program
	{
		static async Task Main(string[] args)
		{
			var configuration = new ConfigurationBuilder()
				.AddJsonFile("appsettings.json", optional: false)
				.Build();
			var dbContextOptions = new DbContextOptionsBuilder<AbpDbContext>()
				.UseSqlServer(configuration.GetConnectionString("Default"))
				.UseLoggerFactory(TestUtils.LoggerFactory)
				.Options;
			using var dbContext = new AbpDbContext(dbContextOptions);
			var users = await dbContext.AbpUsers.OrderBy(p=>p.Id).Skip(10).Take(100).ToListAsyncEF();
			Console.WriteLine(users.Count);

			LinqToDBForEFTools.Initialize();
			using var db = dbContext.CreateLinqToDbContext();
			var queryable = db.GetTable<AbpUser>().OrderBy(p=>p.Id).Skip(10).Take(100);
			var linqUsers = await queryable.ToListAsyncLinqToDB();
			Console.WriteLine(linqUsers.Count);
			var tempUsers = await db.CreateTempTableAsync(queryable, "#TempAbpUsers");
			var result = await tempUsers.ToListAsyncLinqToDB();
			Console.WriteLine(result.Count);
			Console.WriteLine("Hello World!");
		}
	}

	public class AbpDbContext : DbContext
	{
		public DbSet<AbpUser> AbpUsers { set; get; }

		public AbpDbContext(DbContextOptions<AbpDbContext> options) : base(options)
		{

		}

		protected override void OnModelCreating(ModelBuilder modelBuilder)
		{
			modelBuilder.Entity<AbpUser>(p =>
			{
				p.HasKey(p => p.Id);
				p.Property<bool>("IsDeleted").IsRequired();
				p.HasQueryFilter(p => !EF.Property<bool>(p, "IsDeleted"));
			});

			base.OnModelCreating(modelBuilder);
		}
	}

	public class AbpUser
	{
		/// <summary>
		/// ID
		/// </summary>
		public Guid Id { set; get; }

		/// <summary>
		/// UserName
		/// </summary>
		public string UserName { set; get; }
	}
}
```

### Steps to reproduce

Execute similar test code above.

### Environment details

```
Linq2Db: 3.4.2
linq2db.EntityFrameworkCore: 5.4.0
Database Server: SQL Server 2019
Database Provider: EF Core 5.0.2
Operating system: Windows 10
Framework version: .NET 5
```

Please tell me What should I do? Thanks

**Comment by jods4** [comment]
Not sure if it's related or not, but I found a bug in `Sql.Property` recently:
https://github.com/linq2db/linq2db/issues/3158

**Comment by sdanyliv** [comment]
@jods4, no, it is not related. I forgot to convert QueryFilter expression.

**Comment by NetChenQian** [comment]
Thanks, Is there any plan to solve this problem

**Comment by sdanyliv** [comment]
This Thursday I'll release new packages.

**Comment by NetChenQian** [comment]
Nice，thanks

**Comment by sdanyliv** [comment]
Released in 5.5.0
